src/main/java/de/tum/in/www1/artemis/config/CacheConfiguration.java

```diff
@@ -41,128 +43,130 @@ public HibernatePropertiesCustomizer hibernatePropertiesCustomizer(javax.cache.C
+            createIfNotExists(cm, TutorParticipation.class.getName() + ".trainedExampleSubmissions", jcacheConfiguration);
+            createIfNotExists(cm, ExampleSubmission.class.getName(), jcacheConfiguration);
+            createIfNotExists(cm, Exercise.class.getName() + ".tutorParticipations", jcacheConfiguration);
+            createIfNotExists(cm, Course.class.getName() + ".lectures", jcacheConfiguration);
+            createIfNotExists(cm, Course.class.getName() + ".tutorGroups", jcacheConfiguration);
+            createIfNotExists(cm, Exercise.class.getName() + ".attachments", jcacheConfiguration);
+            createIfNotExists(cm, StudentQuestion.class.getName(), jcacheConfiguration);
+            createIfNotExists(cm, StudentQuestion.class.getName() + ".answers", jcacheConfiguration);
+            createIfNotExists(cm, StudentQuestionAnswer.class.getName(), jcacheConfiguration);
+            createIfNotExists(cm, TutorGroup.class.getName(), jcacheConfiguration);
+            createIfNotExists(cm, TutorGroup.class.getName() + ".students", jcacheConfiguration);
+            createIfNotExists(cm, Notification.class.getName(), jcacheConfiguration);
+            createIfNotExists(cm, SystemNotification.class.getName(), jcacheConfiguration);
+            createIfNotExists(cm, GroupNotification.class.getName(), jcacheConfiguration);
+            createIfNotExists(cm, SingleUserNotification.class.getName(), jcacheConfiguration);
+            createIfNotExists(cm, Lecture.class.getName(), jcacheConfiguration);
+            createIfNotExists(cm, Lecture.class.getName() + ".attachments", jcacheConfiguration);
+            createIfNotExists(cm, Attachment.class.getName(), jcacheConfiguration);
+            createIfNotExists(cm, Exercise.class.getName() + ".studentQuestions", jcacheConfiguration);
+            createIfNotExists(cm, Lecture.class.getName() + ".studentQuestions", jcacheConfiguration);
+            createIfNotExists(cm, ModelAssessmentConflict.class.getName(), jcacheConfiguration);
+            createIfNotExists(cm, ModelAssessmentConflict.class.getName() + ".resultsInConflict", jcacheConfiguration);
+            createIfNotExists(cm, ConflictingResult.class.getName(), jcacheConfiguration);
+            createIfNotExists(cm, ExampleSubmission.class.getName() + ".tutorParticipations", jcacheConfiguration);
+            createIfNotExists(cm, de.tum.in.www1.artemis.domain.ProgrammingExerciseTestCase.class.getName(), jcacheConfiguration);
+            // createIfNotExists(cm, );thipster-needle-ehcache-add-entry
+            createIfNotExists(cm, "files", jcacheConfiguration);
         };
     }
 
-    private void createCache(javax.cache.CacheManager cm, String cacheName) {
-        javax.cache.Cache<Object, Object> cache = cm.getCache(cacheName);
-        if (cache != null) {
-            cm.destroyCache(cacheName);
+    // This method is a hotfix for the issue described in: https://github.com/jhipster/generator-jhipster/issues/5354.
```

Comment:
I know it's not a perfect solution, but I did not find a better way.
Appearently this is a known issue in spring boot.
When using a MockBean (mock a service) spring boot tries to set up the same cache twice.
If I disable the cache in the test/application.yml, spring boot can't start because it expects a cache.

Reply:
I also tried using a test profile on the CacheConfiguration, but this caused the same issue as deactivating the cache.